src/Controller/Api/JoinProgrammeController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Api;
+
+use App\Repository\ProgrammeRepository;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+class JoinProgrammeController extends AbstractController
+{
+    private ProgrammeRepository $programmeRepository;
+
+    private UserRepository $userRepository;
+
+    public function __construct(
+        ProgrammeRepository $programmeRepository,
+        UserRepository $userRepository
+    ) {
+        $this->programmeRepository = $programmeRepository;
+        $this->userRepository = $userRepository;
+    }
+
+    /**
+     * @Route (path="/api/programmes/join/{programmeId}", methods={"POST"})
+     */
+    public function index(Request $request, int $programmeId): Response
+    {
+        $loggedInUserId = $this->getUser()->getId();
+        $userToBeJoinedId = json_decode($request->getContent())->id ?? null;
+        $programmeToBeJoined = $this->programmeRepository->findOneBy(['id' => (int)$programmeId]);
+
+        if (null !== $userToBeJoinedId && $loggedInUserId !== $userToBeJoinedId) {
```

Comment:
you are very permissive with your endpoint here. so you either allow the logged in user to join or a specific user if not. your body may or may not contain an id. this is not great from my point of view. that id should be there in the body no matter if I am an admin or not